core/src/main/scala/org/apache/spark/internal/config/ConfigEntry.scala

```diff
@@ -89,14 +89,16 @@ private[spark] abstract class ConfigEntry[T] (
   def defaultValueString: String
 
   protected def readString(reader: ConfigReader): Option[String] = {
-    val values = Seq(
-      prependedKey.flatMap(reader.get(_)),
-      alternatives.foldLeft(reader.get(key))((res, nextKey) => res.orElse(reader.get(nextKey)))
-    ).flatten
-    if (values.nonEmpty) {
-      Some(values.mkString(prependSeparator))
-    } else {
-      None
+    // SPARK-48678: performance optimization: this code could be expressed more succinctly
+    // using flatten and mkString, but doings so adds lots of Scala collections perf. overhead.
```

Comment:
just happened to see ..

```suggestion
    // using flatten and mkString, but doing so adds lots of Scala collections perf. overhead.
```